v2/pkg/engine/resolve/resolve.go

```diff
@@ -412,11 +420,58 @@ func (r *Resolver) handleEvent(event subscriptionEvent) {
 		r.handleTriggerInitialized(event.triggerID)
 	case subscriptionEventKindTriggerShutdown:
 		r.handleTriggerShutdown(event)
+	case subscriptionEventKindHeartbeat:
+		r.handleHeartbeat(event.triggerID, event.data)
 	case subscriptionEventKindUnknown:
 		panic("unknown event")
 	}
 }
 
+func (r *Resolver) handleHeartbeat(id uint64, data []byte) {
+	trig, ok := r.triggers[id]
+	if !ok {
+		return
+	}
+	if r.options.Debug {
+		fmt.Printf("resolver:heartbeat:%d\n", id)
+	}
+	wg := &sync.WaitGroup{}
+	trig.inFlight = wg
+	for c, s := range trig.subscriptions {
```

Comment:
Ideally, I'd love to only send the heartbeat for one particular subscription on a trigger, but I couldn't find an example where we send a subscription ID along.

Reply:
How would this work when we trigger a heartbeat from the handler that distributes to all subscriptions? If this becomes ever a bottleneck, we could track separately what subscription actually need a heartbeat.